main.go

```diff
@@ -1,135 +1,85 @@
-	"github.com/urfave/cli"
-	"k8s.io/client-go/discovery"
-	"k8s.io/client-go/kubernetes"
+	command "github.com/rancher/wrangler-cli"
+	wcrd "github.com/rancher/wrangler/pkg/crd"
+	_ "github.com/rancher/wrangler/pkg/generated/controllers/apiextensions.k8s.io"
+	_ "github.com/rancher/wrangler/pkg/generated/controllers/networking.k8s.io"
+	"github.com/rancher/wrangler/pkg/kubeconfig"
+	"github.com/rancher/wrangler/pkg/ratelimit"
+	"github.com/spf13/cobra"
 	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/klog"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
-func main() {
-	app := cli.NewApp()
-	app.Name = "helm-controller"
-	app.Version = version.FriendlyVersion()
-	app.Usage = "Helm Controller, to help with Helm deployments. Options kubeconfig or masterurl are required."
-	app.Flags = []cli.Flag{
-		cli.StringFlag{
-			Name:   "kubeconfig, k",
-			EnvVar: "KUBECONFIG",
-			Value:  "",
-			Usage:  "Kubernetes config files, e.g. $HOME/.kube/config",
-		},
-		cli.StringFlag{
-			Name:   "master, m",
-			EnvVar: "MASTERURL",
-			Value:  "",
-			Usage:  "Kubernetes cluster master URL.",
-		},
-		cli.StringFlag{
-			Name:   "namespace, n",
-			EnvVar: "NAMESPACE",
-			Value:  "",
-			Usage:  "Namespace to watch, empty means it will watch CRDs in all namespaces.",
-		},
-		cli.IntFlag{
-			Name:   "threads, t",
-			EnvVar: "THREADS",
-			Value:  2,
-			Usage:  "Threadiness level to set, defaults to 2.",
-		},
-	}
-	app.Action = run
+var (
+	debugConfig command.DebugConfig
+)
 
-	if err := app.Run(os.Args); err != nil {
-		klog.Fatal(err)
-	}
+type HelmController struct {
+	Kubeconfig string `short:"k" usage:"Kubernetes config files, e.g. $HOME/.kube/config" env:"KUBECONFIG"`
+	MasterURL  string `short:"m" usage:"Kubernetes cluster master URL" env:"MASTERURL"`
+	Namespace  string `short:"n" usage:"Namespace to watch, empty means it will watch CRDs in all namespaces." env:"NAMESPACE"`
+	Threads    int    `short:"t" usage:"Threadiness level to set, defaults to 2." default:"2" env:"THREADS"`
 }
 
-func run(c *cli.Context) error {
-	masterURL := c.String("master")
-	kubeconfig := c.String("kubeconfig")
-	namespace := c.String("namespace")
-	threadiness := c.Int("threads")
-
-	if threadiness <= 0 {
-		klog.Infof("Can not start with thread count of %d, please pass a proper thread count.", threadiness)
-		return nil
-	}
-
-	klog.Infof("Starting helm controller with %d threads.", threadiness)
-
-	if namespace == "" {
-		klog.Info("Starting helm controller with no namespace.")
-	} else {
-		klog.Infof("Starting helm controller in namespace: %s.", namespace)
-	}
-
-	ctx := signals.SetupSignalHandler(context.Background())
-
-	cfg, err := clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
-	if err != nil {
-		klog.Fatalf("Error building config from flags: %s", err.Error())
-	}
+func (a *HelmController) Run(cmd *cobra.Command, args []string) error {
+	go func() {
+		log.Println(http.ListenAndServe("localhost:6060", nil))
```

Comment:
Is there any way to disable or otherwise configure this listener (set bind address/port, etc)? Is it of any use if debug isn't enabled? Can we add a comment regarding what this listener is for - I'm guessing all that ends up wired into is is the pprof endpoint? Normally enabling pprof is hidden behind a dedicated flag, as it's not something you want enabled for production use.

Reply:
Also, I feel like adding the pprof stuff should be in its own commit at the very least. It's also not called out anywhere in the PR comment or title.